tests/Functional/Schema/ColumnCommentTest.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Tests\Functional\Schema;
+
+use Doctrine\DBAL\Schema\Comparator;
+use Doctrine\DBAL\Schema\Table;
+use Doctrine\DBAL\Tests\FunctionalTestCase;
+use function array_merge;
+use function sprintf;
+
+class ColumnCommentTest extends FunctionalTestCase
+{
+    /** @var bool */
+    private static $initialized = false;
+
+    protected function setUp() : void
+    {
+        parent::setUp();
+
+        if (self::$initialized) {
+            return;
+        }
+
+        self::$initialized = true;
+
+        $table = new Table('column_comments');
+        $table->addColumn('id', 'integer');
+
+        foreach (self::columnProvider() as [$name, $type, $options]) {
+            $table->addColumn($name, $type, $options);
+        }
+
+        $this->connection->getSchemaManager()
+            ->dropAndCreateTable($table);
+    }
+
+    /**
+     * @param array<string,mixed> $options
+     *
+     * @dataProvider columnProvider
+     */
+    public function testColumnComment(string $name, string $type, array $options) : void
+    {
+        $this->assertColumnComment('column_comments', $name, $options['comment'] ?? '');
+    }
+
+    /**
+     * @return mixed[][]
```

Comment:
[Phpstan is ok with `mixed[][]`](https://phpstan.org/r/0c35f6eb-edf4-4cf9-84a0-0162d19105f8), but [Psalm isn't](https://psalm.dev/r/15456cc5ea).

```suggestion
     * @return iterable<mixed[]>
```

Reply:
As far as I understand, `[]` is a list/collection. It doesn't imply a specific implementation. It looks like Psalm thinks differently (cannot find a reference).

I'm fine to adjust but I'd rather do it for the entire codebase and have the rule enforced on CI. E.g. by adding a dependency on Psalm and setting the lowest possible level. Otherwise, it will become a matter of personal preference than a standard.

Reply:
Ok let's not make this a blocker